packages/flutter/lib/src/material/tabs.dart

```diff
@@ -395,9 +395,11 @@ class _IndicatorPainter extends CustomPainter {
 
     if (indicatorSize == TabBarIndicatorSize.label) {
       final double tabWidth = tabKeys[tabIndex].currentContext!.size!.width;
-      final double delta = ((tabRight - tabLeft) - tabWidth) / 2.0;
-      tabLeft += delta;
-      tabRight -= delta;
+      final EdgeInsetsGeometry labelPadding = tabKeys[tabIndex].currentContext!.findAncestorRenderObjectOfType<RenderPadding>()!.padding;
+      final EdgeInsets insets = labelPadding.resolve(_currentTextDirection);
```

Comment:
Call `findAncestorRenderObjectOfType` to find the `RenderPadding` looks a little odd.
What will hanppen if  the `Padding` widget remove in the future?

Reply:
1. label padding is the padding outsize of label, if you put it into KeyedSubtree, the size from key is also change, you can't clarify what is the TabBarIndicatorSize.label.
2. if someone remove Padding, it will be a big breaking change for the customer, and test case will not be passed. and we are in nullsafety,  'EdgeInsetsGeometry? labelPadding ' should not be, if we already know it.

Reply:
Do we have a better way to get the label padding info?

Reply:
i have no better idea for now 😳

Reply:
Reaching out to the RenderObject does not look right. If this padding is controlled by the TabBar, we should pass it directly into the painter.